### PR TITLE
📖 Update BMH states that are able to consume detach annotation

### PR DESCRIPTION
**Pull request by furkatgofurov7**
Updates BMH states list since [967](https://github.com/metal3-io/baremetal-operator/pull/967) and [965](https://github.com/metal3-io/baremetal-operator/pull/965) were introduced.

### CODE REVIEW

File: design/baremetal-operator/detached-annotation.md [hunk 1]
```diff
@@ -78,8 +78,8 @@ baremetalhost.metal3.io/detached: ""
 The value is ignored, similar to the `paused` annotation and could optionally
 include context from the system/user which applies the annotation.
 
-This annotation will only be consumed while the BMH is in either `Provisioned`
-or `ExternallyProvisioned` state, in all other cases it is ignored.
+This annotation will only be consumed while the BMH is in either `Provisioned`,
+`ExternallyProvisioned` or `Ready`/`Available` state, in all other cases it is ignored.
```
Comment (fmuyassarov) [inline design/baremetal-operator/detached-annotation.md:82]: `Available` state is not implemented yet, no?

Comment (furkatgofurov7) [reply design/baremetal-operator/detached-annotation.md:82]: This is based on this [comment](https://github.com/metal3-io/baremetal-operator/pull/967#discussion_r703677051)

**Review by kashifest — APPROVED** [review]

**Comment by furkatgofurov7** [comment]
/unhold
967 is merged.

**Pull request merged**